components/brave_wallet_ui/page/async/wallet_page_async_handler.ts

```diff
@@ -218,4 +226,36 @@ handler.on(WalletPageActions.openWalletSettings.getType(), async (store) => {
   })
 })
 
+handler.on(WalletPageActions.getNFTMetadata.getType(), async (store, payload: BraveWallet.BlockchainToken) => {
+  store.dispatch(WalletPageActions.setIsFetchingNFTMetadata(true))
+  const { jsonRpcService } = getAPIProxy()
+  const result = await jsonRpcService.getERC721Metadata(payload.contractAddress, payload.tokenId, payload.chainId)
+
+  if (!result.error) {
+    const response = JSON.parse(result.response)
+    const tokenNetwork = getTokensNetwork(getWalletState(store).networkList, payload)
+    const nftMetadata: NFTMetadataReturnType = {
+      chainName: tokenNetwork.chainName,
+      tokenType: 'ERC721', // getNFTMetadata currently supports only ERC721 standard. When other standards are supported, this value should be dynamic
+      tokenID: payload.tokenId,
+      imageURL: response.image.startsWith('data:image/') ? response.image : httpifyIpfsUrl(response.image),
+      floorFiatPrice: '',
+      floorCryptoPrice: '',
+      contractInformation: {
+        address: payload.contractAddress,
+        name: response.name,
+        description: response.description,
+        website: '',
+        twitter: '',
+        facebook: '',
+        logo: ''
+      }
+    }
+    store.dispatch(WalletPageActions.updateNFTMetadata(nftMetadata))
+  } else {
+    console.error(result.errorMessage)
```

Comment:
This doesn't seem to be properly throwing an error here when I set my NFT URL to `ipfs%3A%2F%2Fbafybeiflkwnmrlg45dhkg3y35ajiiflspnuqtyz6goq4iism2eo43ciiva%2Fface_parts%2FAsset%20331.svg`

in this `updateTokenURI` call (which appears to be causing issues with the JSON parser of and throwing a `-32700` error code from the provider). Here's what I'm seeing in console:
![Screen Shot 2022-07-05 at 3 21 16 PM](https://user-images.githubusercontent.com/23125059/177243330-28f69329-938a-4441-953f-c517effdb148.png)
![Screen Shot 2022-07-05 at 3 21 52 PM](https://user-images.githubusercontent.com/23125059/177243339-529078ff-f676-4811-9aa0-d53f062cfce1.png)

Is this is the expected behavior of the screen when it's unable to load the NFT url?

Reply:
@nvonpentz Can you clarify this?

Reply:
@kdenhartog It seems you got it to work in the latest build, any actions you would like us to work on?